datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -216,6 +218,41 @@ impl CoalesceBatchesStream {
             match input_batch {
                 Poll::Ready(x) => match x {
                     Some(Ok(batch)) => {
+                        let new_columns: Vec<Arc<dyn Array>> = batch
+                            .columns()
+                            .iter()
+                            .map(|c| {
+                                // Try to re-create the `StringViewArray` to prevent holding the underlying buffer too long.
+                                if let Some(s) = c.as_string_view_opt() {
+                                    let view_cnt = s.views().len();
+                                    let buffer_size = s.get_buffer_memory_size();
+
+                                    // Re-creating the array copies data and can be time consuming.
+                                    // We only do it if the array is sparse, below is a heuristic to determine if the array is sparse.
+                                    if buffer_size > (view_cnt * 32) {
```

Comment:
If there are lots of long strings in the buffer, will it be triggered every time even if nothing has been filtered?

Reply:
Yes, this is a rough heuristic to test whether the buffer is dense.
In fact, there's no way for us to know if anything has been filtered in `CoalesceBatchesExec` -- we only know this in downstream operators like `FilterExec`. The fact that we have a `CoalesceBatchesExec` here means that it is likely that the batch has been sparse and thus requires gc.